st2api/st2api/controllers/v1/keyvalue.py

```diff
@@ -395,7 +478,19 @@ def delete(self, name, requester_user, scope=FULL_SYSTEM_SCOPE, user=None):
             )
 
             try:
-                KeyValuePair.delete(kvp_db)
+                if scope in [SYSTEM_SCOPE, FULL_SYSTEM_SCOPE]:
+                    if is_admin:
+                        KeyValuePair.delete(kvp_db)
+                    else:
+                        for key in get_all_system_kvp_names_for_user(user):
+                            try:
+                                if key == name:
+                                    KeyValuePair.delete(kvp_db)
+                            except Exception as e:
+                                LOG.error("Unable to get key %s: %s", key, str(e))
```

Comment:
Why is this in a try/except block? Won't the outer try/except block catch this?

Also, I think we need a `break`.
```suggestion
                            if key == name:
                                kvp_db = KeyValuePair.add_or_update(kvp_db)
                                break
```

If we do need the try/except block, then the error message needs to change:
```suggestion
                            try:
                                if key == name:
                                    kvp_db = KeyValuePair.add_or_update(kvp_db)
                            except Exception as e:
                                LOG.error("Unable to delete key %s: %s", key, str(e))
```